src/services/utilities.ts

```diff
@@ -1358,4 +1358,128 @@ namespace ts {
             diagnostics: error ? concatenate(diagnostics, [error]) : diagnostics
         };
     }
-}
+
+    export function getMissingAbstractMemberInsertion(classDecl: ClassDeclaration, checker: TypeChecker, newlineChar: string): string {
+        const classSymbol = checker.getSymbolOfNode(classDecl);
+
+        const InstantiatedExtendsType = <InterfaceType>checker.getTypeFromTypeReference(getClassExtendsHeritageClauseElement(classDecl));
+        const resolvedExtendsType = checker.resolveStructuredTypeMembers(InstantiatedExtendsType);
+
+        const missingMembers = filterMissingMembers(filterAbstract(filterNonPrivate(resolvedExtendsType.members)), classSymbol.members);
+
+        return getInsertionsForMembers(missingMembers, classDecl, checker, newlineChar);
+    }
+
+    export function getMissingInterfaceMembersInsertion(classDecl: ClassDeclaration, checker: TypeChecker, newlineChar: string): string {
+        const implementedTypeNodes = getClassImplementsHeritageClauseElements(classDecl);
+
+        const classSymbol = checker.getSymbolOfNode(classDecl);
+
+        let implementsIntersectionType: IntersectionType | InterfaceType;
+        if (implementedTypeNodes.length > 1) {
+            implementsIntersectionType = <IntersectionType>checker.getIntersectionType(implementedTypeNodes.map(checker.getTypeFromTypeReference));
+        }
+        else {
+            implementsIntersectionType = <InterfaceType>checker.getTypeFromTypeReference(implementedTypeNodes[0]);
+        }
+
+        const structuredType = checker.resolveStructuredTypeMembers(<IntersectionType | InterfaceType>implementsIntersectionType);
+        const missingMembers = filterMissingMembers(filterNonPrivate(structuredType.members), classSymbol.members);
+        return getInsertionsForMembers(missingMembers, classDecl, checker, newlineChar);
+    }
+
+    /**
+     * Finds the symbols in source but not target, generating a new map with the differences.
+     */
+    function filterMissingMembers(sourceSymbols: Map<Symbol>, targetSymbols: Map<Symbol>): Map<Symbol> {
+        const result: Map<Symbol> = createMap<Symbol>();
+        outer:
+        for (const sourceName in sourceSymbols) {
+            for (const targetName in targetSymbols) {
+                if (sourceName === targetName) {
+                    continue outer;
+                }
+            }
+            result[sourceName] = sourceSymbols[sourceName];
+        }
+        return result;
+    }
+
+    function filterSymbolMapByDecl(symbolMap: Map<Symbol>, pred: (decl: Declaration) => boolean): Map<Symbol> {

```

Comment:
I think it is ok here to write your own loop as the logic is not very complicated